checkers/invalid_range_index_checker.py

```diff
@@ -0,0 +1,51 @@
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from ast import literal_eval
+
+
+class InvalidRangeIndexChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = 'custom'
+    msgs = {'E9993':
+                ('You should not use invalid range index on line %s',
+                 'invalid-range-index',
+                 'Used when you use invalid index range')}
+    options = ()
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("invalid-range-index")
+    def visit_call(self, node):
+        if isinstance(node.func, astroid.Name):
+            name = node.func.name
+            # ignore the name if it's not a builtin (i.e. not defined in the
+            # locals nor globals scope)
+            if not ((name in node.frame() or name in node.root()) and
+                            name == 'range'):
+                # the arguments of 'range' call
+                arg = node.args
+                eval_parm = list(map(lambda z: literal_eval(z.as_string()), arg))
```

Comment:
@Jasminewwwu I realized after I merged this in that this line will raise an error if the range input is not a constant. I have disabled this checker until this problem is fixed.

In general, we should never use `eval`, so you need to find a new approach.